src/devices/Ft232H/Ftx232HDevice.cs

```diff
@@ -0,0 +1,1074 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Iot.Device.Ft232H;
+using Iot.Device.Board;
+using Iot.Device.Ft2232H;
+using Iot.Device.Ft4232H;
+
+namespace Iot.Device.FtCommon
+{
+    /// <summary>
+    /// FTx32H base Device
+    /// </summary>
+    public class Ftx232HDevice : FtDevice, IDisposable
+    {
+        private const int PinNumberFT2x = 16;
+        private const int PinNumberFT4x = 8;
+
+        /// <summary>
+        /// Number of pins
+        /// </summary>
+        internal const int PinCountConst = 16;
+
+        // Commands are available in:
+        // - AN_108_Command_Processor_for_MPSSE_and_MCU_Host_Bus_Emulation_Modes.pdf
+        // - AN_113_FTDI_Hi_Speed_USB_To_I2C_Example.pdf
+        // To understand the signal need for I2C communication: https://training.ti.com/sites/default/files/docs/slides-i2c-protocol.pdf
+        private const uint I2cMasterFrequencyKbps = 400;
+        // Direction values
+        private const byte I2cDirSDAinSCLin = 0x00;
+        private const byte I2cDirSDAinSCLout = 0x01;
+        private const byte I2cDirSDAoutSCLin = 0x02;
+        private const byte I2cDirSDAoutSCLout = 0x03;
+        // Data values
+        private const byte I2cDataSDAloSCLhi = 0x01;
+        private const byte I2cDataSDAhiSCLhi = 0x03;
+        private const byte I2cDataSDAloSCLlo = 0x00;
+        private const byte I2cDataSDAhiSCLlo = 0x02;
+        private const byte NumberCycles = 6;
+        private const byte MaskGpio = 0xF8;
+
+        private SafeFtHandle _ftHandle = null!;
+
+        // This is used by FT232H and others to track the GPIO states
+        internal byte GpioLowData = 0;
+        internal byte GpioLowDir = 0;
+        internal byte GpioHighData = 0;
+        internal byte GpioHighDir = 0;
+
+        internal bool[] PinOpen = new bool[PinCountConst];
+        internal PinMode[] GpioDirections = new PinMode[PinCountConst];
+        internal bool UsesMpseeForGpio = true;
+
+        internal List<SpiConnectionSettings> ConnectionSettings = new List<SpiConnectionSettings>();
+
+        /// <summary>
+        /// Gets all the FT232H connected
+        /// </summary>
+        /// <returns>A list of FT232H</returns>
```

Comment:
Comment is wrong (this apparently returns any type of FTxxxx device)

Reply:
And I checked all of them, I forgot to adjust quite some. Thanks!